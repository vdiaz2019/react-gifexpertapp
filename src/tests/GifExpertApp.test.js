import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Debe de funcionar correctamente el componente <GifExpertApp />', () => {

    test('Debe de renderizarse correctamente el componente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('Debe de mostrar una lista de categorÃ­as', () => {
        const categories = ['Samurai X', 'Simpsons'];
        const wrapper = shallow(<GifExpertApp defaultCategories = {categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
})
