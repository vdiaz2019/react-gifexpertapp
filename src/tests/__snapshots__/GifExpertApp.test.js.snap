// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe de funcionar correctamente el componente <GifExpertApp /> Debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Samurai X"
      key="Samurai X"
    />
    <GifGrid
      category="Simpsons"
      key="Simpsons"
    />
  </ol>
</Fragment>
`;

exports[`Debe de funcionar correctamente el componente <GifExpertApp /> Debe de renderizarse correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
